_P101_TMP117.ino

```diff
@@ -0,0 +1,269 @@
+   *    2         float         set temperature offset like "2 20.5" sets the offset to 20.5°C
+   *    3         float         calibrate sensor to target temperature like "3 30.5" will calibrate sensor to 30.5°C
+   * 
+   */
+        // Get sensor resolution configuration
+         uint8_t ConvTime = Settings.TaskDevicePluginConfig[event->TaskIndex][0];
+         uint8_t Averaging = Settings.TaskDevicePluginConfig[event->TaskIndex][1];
+         float adjust = Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2];
+         float offset = Settings.TaskDevicePluginConfigFloat[event->TaskIndex][3];
+         
+        // Initiate wire library and serial communication
+  Wire.begin();           
+  // Serial.begin(115200);
+  /* The default setup is :
+   *    Conversion mode = CONTINUOUS  ---> continuoss
+   *    Conversion time = C125mS      -|
+   *    Averaging mode  = AVE8        -|-> new data every 125mS
+   *    Alert mode      = data        ---> alert pin states that new data is available
+   *    
+   */
+  tmp.init ( NULL );                // no callback
+  tmp.setConvMode (CONTINUOUS);     // contious measurement, also ONESHOT or SHUTWDOWN possible
+
+ switch (ConvTime)
+        {
+        case (0x00):
+          tmp.setConvTime (C15mS5);         // 1. setup C125mS+NOAVE = 15.5 mS measurement time
+          break;
+        case (0x01):
+          tmp.setConvTime (C125mS);         
+          break;
+        case (0x02):
+          tmp.setConvTime (C250mS);
+          break;
+        case (0x03):
+          tmp.setConvTime (C500mS);
+          break;
+        case (0x04):
+          tmp.setConvTime (C1S);
+          break;
+        case (0x05):
+          tmp.setConvTime (C4S);
+          break;
+        case (0x06):
+          tmp.setConvTime (C8S);
+          break;
+        case (0x07):
+          tmp.setConvTime (C16S);
+          break;   
+        }
+
+ switch (Averaging)
+        {
+        case (0x00):
+          tmp.setAveraging (NOAVE); 
+          break;
+        case (0x01):
+          tmp.setAveraging (AVE8);         
+          break;
+        case (0x02):
+          tmp.setAveraging (AVE32);
+          break;
+        case (0x03):
+          tmp.setAveraging (AVE64);
+          break;
+        }       
+
+          tmp.setOffsetTemperature ( offset );      //set temperature offset
+          tmp.setTargetTemperature ( adjust );      //calibrate sensor to target temperature 
+          tmp.update();
+
+            double temperature = tmp.getTemperature();
+            UserVar[event->BaseVarIndex] = temperature;
+                                         //  UserVar[event->BaseVarIndex + 1] = humidity;
+            String log = F("TMP117: Temperature: ");
+            log += UserVar[event->BaseVarIndex];
+            addLog(LOG_LEVEL_INFO, log);
+            
+            success = true;
+      if(!tmp.getDeviceID())
+        {
+         success = false;

```

Comment:
If it doesn't have a device ID, why even try to set the output value first?
Apparently you do set `success` to false, so it will never register a new updated value and thus also never try to refresh it in the devices page.